src/justfile.rs

```diff
@@ -28,16 +28,32 @@ impl<'src> Justfile<'src> {
     self.recipes.len()
   }
 
-  pub(crate) fn suggest(&self, name: &str) -> Option<&'src str> {
-    let mut suggestions = self
+  pub(crate) fn suggest(&self, name: &str) -> Option<(&'src str, Option<&'src str>)> {
+    let mut available: Vec<(&str, usize, Option<&str>)> = self
```

Comment:
This can be cleaned up by chaining iterators, something like:

```
  let mut suggestions = self
    .recipes
    .keys()
    .map(|name| {
      (edit_distance(name, input), Suggestion {
        name,
        target: None,
      })
    })
    .chain(aliases.iter().map(|name, target| {
      (edit_distance(name, input), Suggestion {
        name,
        target: Some(target),
      })
    }))
    .filter(|(distance, _suggestion)| distance < 3)
    .collect::<Vec<Suggestion>>();

  suggestions.sort_by_key(|(distance, _suggestion)| distance);

  suggestions.first()
```

Reply:
Ah, thanks for putting me onto `.chain`!